src/components/NotFound.jsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import withSideEffect from 'react-side-effect';
+
+/**
+ * Use this component as a wrapper for your 'not found' UI when the API returns
+ * no data
+ *
+ * Note that only one child is allowed for this component since it does not
+ * provide any rendered output itself
+ */
+class NotFound extends React.Component {
+	render() {
+		if (this.props.children) {
+			return React.Children.only(this.props.children);
+		}
+		return null;
+	}
+}
+
+function reducePropsToState(propsList) {
+	if (!propsList.length) {
+		return;
+	}
+	return 404;
+}
+
+function handleStateChangeOnClient(code) {
```

Comment:
it is. `withSideEffect` needs to know what to do when the app gets rendered on the server and then what happens when it gets re-rendered on the client, but in this case we want to no-op on the client. For better or worse, the package [forces this to be a function](https://github.com/gaearon/react-side-effect/blob/master/src/index.js#L13)